frontend/public/components/pod-logs.jsx

```diff
@@ -1,194 +1,78 @@
 import * as _ from 'lodash-es';
 import * as React from 'react';
 
-import { resourceURL } from '../module/k8s';
-import { PodModel } from '../models';
-import { SafetyFirst } from './safety-first';
-import { Dropdown, LoadingInline, LogWindow, ResourceIcon, TogglePlay, lineBuffer, stream } from './utils';
+import { Dropdown, LoadingInline, ResourceLog, ResourceName } from './utils';
 
-const dataHasFailureMsg = (data) => {
-  return _.includes(data, '"status": "Failure"');
+const isContainerTerminated = (container) => {
+  return !_.isEmpty(_.get(container, 'status.terminated'));
 };
 
-const dataHasHTML = (data) => {
-  return _.includes(data, '<html') || _.includes(data, '<HTML');
+// Component to render resource dropdown or just the resource name if only
+// one resource is provided.
+const ContainerDropdown = ({currentContainer, containers, kind, onChange}) => {
+  const currentResourceNameWithIcon = <ResourceName name={currentContainer.name || <LoadingInline />} kind={kind} />;
+
+  if (containers.length > 1) {
+    const dropdownItems = _.map(containers, ({name}) => {
+      return <ResourceName name={name} kind={kind} />;
+    });
+    return <Dropdown className="btn-group" items={dropdownItems} title={currentResourceNameWithIcon} onChange={onChange} />;
+  }
+  return currentResourceNameWithIcon;
 };
 
-export class PodLogs extends SafetyFirst {
+export class PodLogs extends React.Component {
   constructor(props) {
     super(props);
-
-    this._buffer = lineBuffer(1000);
-    this._loadTime = Date.now();
-    this._pendingReload = null;
-
-    this._touchLoadTimeState = _.throttle(this._touchLoadTimeState, 100);
     this._selectContainer = this._selectContainer.bind(this);
-    this._updateLogState = this._updateLogState.bind(this);
-    this._toggleLogState = this._toggleLogState.bind(this);
 
     this.state = {
-      containerNames: [],
-      currentContainer: '',
-      logURL: '',
-      loadTime: 0,
-      logState: 'loading'
+      containers: [],
+      currentContainer: {},
```

Comment:
It looks like we only ever reference `currentContainer.name` - can we just track that instead?

Reply:
On line 52 and 58 I'm using currentContainer.status to check if the container is terminated.